packages/sanity/src/core/studio/StudioLayout.tsx

```diff
@@ -18,6 +18,14 @@ import {
 import {StudioErrorBoundary} from './StudioErrorBoundary'
 import {useWorkspace} from './workspace'
 
+const DetectViteDevServerStopped = lazy(() =>
+  import('./ViteDevServerStopped').then((DevServerStopped) => ({
+    default: DevServerStopped.DetectViteDevServerStopped,
+  })),
+)
+
+const detectViteDevServerStopped = import.meta.hot && process.env.NODE_ENV === 'development'
```

Comment:
@juice49 tightened the conditional here to only being for vite servers (`import.meta.hot` would typically only exist for vite). What do you think?

Reply:
I guess other tooling beyond Vite could implement `import.meta.hot`, but it seems like a pretty good indication that the environment supports HMR to me 🙂.